tests/chainer_tests/functions_tests/pooling_tests/test_max_pooling_nd.py

```diff
@@ -0,0 +1,199 @@
+            for xs in xss]
+
+
+@testing.parameterize(*testing.product({
+    'dims': [(4,), (4, 3), (4, 3, 2)],
+    'cover_all': [True, False],
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
+class TestMaxPoolingND(unittest.TestCase):
+
+    def setUp(self):
+        self.ndim = len(self.dims)
+        self.ksize = (3,) * self.ndim
+        self.stride = (2,) * self.ndim
+        self.pad = (1,) * self.ndim
+
+        # Avoid unstability of numerical gradient
+        x_shape = (2, 3) + self.dims
+        self.x = numpy.arange(
+            functools.reduce(mul, x_shape), dtype=self.dtype).reshape(x_shape)
+        self.x = 2 * self.x / self.x.size - 1
+
+        outs = tuple(conv.get_conv_outsize(d, k, s, p, self.cover_all)
+                     for (d, k, s, p)
+                     in zip(self.dims, self.ksize, self.stride, self.pad))
+        gy_shape = (2, 3) + outs
+        self.gy = numpy.random.uniform(-1, 1, gy_shape).astype(self.dtype)
+
+        self.check_backward_options = {'eps': 2.0 ** -8}
+        if self.dtype == numpy.float16:
+            self.check_backward_options = {
+                'eps': 2.0 ** -8, 'atol': 1e-03, 'rtol': 1e-03}
+
+    def check_forward(self, x_data, use_cudnn=True):
+        dims = self.dims
+        ksize = self.ksize
+        stride = self.stride
+        pad = self.pad
+        x = chainer.Variable(x_data)
+        y = functions.max_pooling_nd(x, ksize, stride=stride, pad=pad,
+                                     cover_all=self.cover_all,
+                                     use_cudnn=use_cudnn)
+        self.assertEqual(y.data.dtype, self.dtype)
+        y_data = cuda.to_cpu(y.data)
+
+        self.assertEqual(self.gy.shape, y_data.shape)
+        for k in six.moves.range(2):
+            for c in six.moves.range(3):
+                x = self.x[k, c]
+                expect = numpy.array(
+                    [x[idx].max() for idx in pooling_patches(
+                        dims, ksize, stride, pad, self.cover_all)]
+                ).reshape(y_data.shape[2:])
+                testing.assert_allclose(expect, y_data[k, c])
+
+    @condition.retry(3)
+    def test_forward_cpu(self):
+        self.check_forward(self.x, use_cudnn=False)
+
+    def test_forward_cpu_wide(self):  # see #120
+        ndim = self.ndim
+        x_shape = (2, 3) + (15,) * ndim
+        x_data = numpy.random.rand(*x_shape).astype(self.dtype)
+        x = chainer.Variable(x_data)
+        ksize = stride = int(math.ceil(pow(32, 1.0 / ndim)))
+        functions.max_pooling_nd(x, ksize, stride=stride, pad=0)
+
+    @attr.cudnn
+    @condition.retry(3)
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x))
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_forward_gpu_no_cudnn(self):
+        self.check_forward(cuda.to_gpu(self.x), False)
+
+    def test_forward_consistency_regression(self):
+        # Regression test to max_pooling_2d.
+
+        if len(self.dims) != 2:
```

Comment:
I suppose that `max_pooling_2d` does not work with `self.dims` other than two, so the result of `max_pooling_nd` can not be tested with that of `max_pooling_2d`.